### PR TITLE
Implement `AINT`

### LINKED ISSUE #2343: Implement `AINT`
```fortran
program main
real :: x
x = 1.23
print *, aint(x)
end program
```

> LFortran

```fortran
% lfortran a.f90 
semantic error: Function 'aint' not found or not implemented yet (if it is intrinsic)
 --> a.f90:4:10
  |
4 | print *, aint(x)
  |          ^^^^^^^ 


Note: Please report unclear or confusing messages as bugs at
https://github.com/lfortran/lfortran/issues.
```

> GFortran

```console
 gfortran a.f90 && ./a.out
   1.00000000    
```

### PR DESCRIPTION
**Pull request by Pranavchiku**
Fixes #2343

### CODE REVIEW

File: src/runtime/builtin/lfortran_intrinsic_builtin.f90 [hunk 1]
```diff
@@ -56,6 +56,11 @@ integer function int(x, kind)
     integer, optional, intent(in) :: kind
     end function
 
+    real function aint(x, kind)
+    real, intent(in) :: x(:)
+    integer, optional, intent(in) :: kind
+    end function
```
Comment (certik) [inline src/runtime/builtin/lfortran_intrinsic_builtin.f90:62]: New intrinsics should got into the IntrinsicFunction node. We need to move more functions from this file into ASR.

Comment (Pranavchiku) [reply src/runtime/builtin/lfortran_intrinsic_builtin.f90:62]: I initially thought of implementing it as intrinsic function, but when I saw `int` just adds a casting node at runtime, so I thought of having a similar implementation for this.

File: src/lfortran/semantics/comptime_eval.h [hunk 2]
```diff
@@ -801,6 +802,43 @@ TRIG2(sqrt, dsqrt)
...
+                int64_t ival = ASR::down_cast<ASR::IntegerConstant_t>(ASRUtils::expr_value(real_expr))->m_n;
+                float fval = static_cast<float>(ival);
+                return ASR::down_cast<ASR::expr_t>(ASR::make_RealConstant_t(al, loc, fval, real_type));
+            } else {
+                int64_t ival = ASR::down_cast<ASR::IntegerConstant_t>(ASRUtils::expr_value(real_expr))->m_n;
+                float fval = static_cast<float>(ival);
+                return ASR::down_cast<ASR::expr_t>(ASR::make_RealConstant_t(al, loc, fval, real_type));
+            }
+        } else if (ASR::is_a<ASR::Real_t>(*real_type)) {
+            std::cout<<"real_kind: "<<real_kind<<std::endl;
```
Comment (Pranavchiku) [inline src/lfortran/semantics/comptime_eval.h:822]: I forgot to remove print statements from here.

File: src/runtime/builtin/lfortran_intrinsic_builtin.f90 (hunk 1 above)
Comment (Pranavchiku) [reply src/runtime/builtin/lfortran_intrinsic_builtin.f90:62]: Done at #2354

**Comment by gptsarthak** [comment]
I think you can close this now.

**Comment by Pranavchiku** [comment]
Closing with reference to #2354.

**Pull request closed without merging**